daml/Marketplace/Trading/Service.daml

```diff
@@ -1,99 +1,85 @@
-
 module Marketplace.Trading.Service where
 
 import DA.Foldable (sequence_)
+import DA.Optional (whenSome)
 import DA.Set qualified as Set
-import DA.Finance.Asset (AssetDeposit)
-import DA.Finance.Asset.Settlement (AssetSettlementRule)
-import DA.Finance.Types (Account)
+import DA.Finance.Asset (AssetDeposit, AssetDeposit_Lock(..), AssetDeposit_Unlock(..))
 import DA.Finance.Utils (fetchAndArchive)
+import Marketplace.Custody.Service qualified as Custody
 import Marketplace.Trading.Model qualified as Order
 import Marketplace.Clearing.Service qualified as Clearing
 import Marketplace.Trading.Confirmation.Model qualified as Confirmation
 import Marketplace.Trading.Error (Error (..))
-import qualified Marketplace.Rule.AllocationAccount as AllocationAccountRule (T, Deposit(..), Withdraw(..))
 import Marketplace.Utils
 
 type T = Service
 
-whenCollateral : Order.TradeCollateral -> (ContractId AssetDeposit -> Update (ContractId AssetDeposit)) -> Update Order.TradeCollateral
-whenCollateral (Order.Cleared x) _ = return $ Order.Cleared x
-whenCollateral (Order.Collateral depositCid) fn = Order.Collateral <$> fn depositCid
-
 template Service
   with
     operator : Party
     provider : Party
     customer : Party
-    tradingAccount : Account
-    allocationAccount : Account
   where
     signatory operator, provider, customer
 
     key (operator, provider, customer) : (Party, Party, Party)
     maintainer key._1
 
     let
-      withdrawFromAllocationAccount : Order.TradeCollateral -> Update Order.TradeCollateral
-      withdrawFromAllocationAccount = \case
-        (Order.Cleared party)         -> return $ Order.Cleared party
-        (Order.Collateral depositCid) -> Order.Collateral <$> exerciseByKey @AllocationAccountRule.T allocationAccount.id AllocationAccountRule.Withdraw with transferTo = tradingAccount, ..
-
-      withdrawFeesFromAllocationAccount : Optional (ContractId AssetDeposit) -> Update (Optional (ContractId AssetDeposit))
-      withdrawFeesFromAllocationAccount = \case
-        None            -> return None
-        Some depositCid -> Some <$> exerciseByKey @AllocationAccountRule.T allocationAccount.id
-                                      AllocationAccountRule.Withdraw with transferTo = tradingAccount, ..
-
+      unlockDeposit : ContractId AssetDeposit -> Update ()
+      unlockDeposit = \depositCid -> exercise depositCid AssetDeposit_Unlock >> pure ()
+      unlockCollateral : Order.MarketType -> Update ()
+      unlockCollateral = \case
+        Order.Collateralized depositCid _ -> unlockDeposit depositCid
+        _ -> pure ()
 
     controller customer can
-      nonconsuming RequestCreateOrder : Either (ContractId Order.T)
-                                               (ContractId Order.T, ContractId CreateOrderRequest, Order.TradeCollateral)
+      nonconsuming RequestCreateOrder : ContractId Order.T
         with
```

Comment:
Any reason why we don't (need to?) return the `ContractId CreateOrderRequest` here anymore? I'm thinking it might come in handy in test scripts, where we need to pass it into the next choice.

Reply:
I just saw further down that in the error case we just create an `Order` contract, which is probably the reason for this change in choice signature. I wonder if we should just `fail` in case something is wrong with the order on entry.

Reply:
(but ^ is also not something to do in this PR)

Reply:
> Any reason why we don't (need to?) return the `ContractId CreateOrderRequest` here anymore? I'm thinking it might come in handy in test scripts, where we need to pass it into the next choice.

I presume it was so we have easy access to it when testing but it was actually never referenced in any testing scripts nor in the entire project which was why I removed it. 

> I just saw further down that in the error case we just create an `Order` contract, which is probably the reason for this change in choice signature. I wonder if we should just `fail` in case something is wrong with the order on entry.

Great point - handling failure/error scenarios is something we need to review in the entire Marketplace. I believe the error cases were added so it would be easy for the investor to see why an order has not been submitted for execution as there was no other way of "nicely" displaying an error message back to the user at that point in time. 

> (but ^ is also not something to do in this PR)

👍